source/agora/network/NetworkManager.d

```diff
@@ -561,6 +565,18 @@ public class NetworkManager
         return false;
     }
 
+    /// Start the StatsServer
+    public void startStatsServer()
+    {
+        stats_server = new StatsServer(node_config.stats_listening_port);
+    }
```

Comment:
Note that for dependency injection, the usual pattern is to `return` the value, not to rely on side effect.

Reply:
Also if `stopStatsServer` is called without a call to `startStatsServer`, there will be a segfault.

Reply:
> stopStatsServer

that would be a programming error though, no one is supposed to call the stopStatsServer without calling startStatsServer